doc/vxlan/Vxlan_kernel_routes.md

```diff
@@ -0,0 +1,426 @@
+                |--- endpoint = 100.100.100.1
+                |--- vni = 2000
+
+--- Kernel
+  |--- Vnet_1000
+         |--- Brvxlan1000  -> A bridge for Vnet that terminates Vxlan and does L2 forwarding
+         |--- Vxlan1000 -> vxlan interface
+```
+
+For the above config, SONiC creates kernel configs for a L2 bridge and a VxLAN interface. For the vxlan routes that are added using `VXLAN_ROUTE_TUNNEL`, there are no kernel configurations applied. The kernel cannot initiate communication to the vnet endpoints behind VTEP since the kernel interface and routes for these prefixes are not installed on the kernel. This document enhances the VxLAN capabilities of SONiC to have the kernel routes and vxlan P2P interface to communicate with the remote endpoints defined in `VNET_ROUTE_TUNNEL`. This can be used for traffic originated by CPU (like BGP, ping etc) and destined to a remote VTEP endpoint. 
+
+# 5 Usecase
+
+Consider a sample vnet configuration as below:
+```
+    "VNET": {
+        "Vnet_test": {
+            "vni": "1000",                  // VNET's VNI is 1000  
+            "vxlan_tunnel": "tunnel_v4"
+        }
+    },
+    "VNET_ROUTE_TUNNEL": {
+        "Vnet_test|20.0.0.2/32": {
+            "endpoint": "200.200.200.2",
+            "vni": "1000"                  // Route uses same VNI as VNET
+        },
+        "Vnet_test|20.0.0.3/32": {
+            "endpoint": "200.200.200.3",
+            "vni": "2000"                  // Route has VNI 2000
+        }
+    }
+```
+
+In the above config, there is a VNET with name `Vnet_test` having a VNI `1000`. When this VNET is created, SONiC creates a linux bridge and vxlan interface on the kernel to encap and decap vxlan packets with VNI 1000. The above config also has two routes in the same VNET: one route to 20.0.0.2 behind a VTEP 200.200.200.2 having VNI 1000 and another route to 20.0.0.3 behind a VTEP having VNI 2000.
+
+Consider a usecase where SONiC has to establish BGP to both the devices: 20.0.0.2 and 20.0.0.3. CPU can initiate traffic to endpoints in this VNET which have the same VNI as the VNET since the kernel routes and interfaces are configured. For example, SONiC can send/receive traffic to 20.0.0.2 VM which is behind VTEP 200.200.200.2 using VNI 1000. So SONiC can establish BGP session with 20.0.0.2 device on the VNET.
+
+However, the VM having IP 20.0.0.3 is behind a VTEP 200.200.200.3 having VNI 2000. SONiC does not have any kernel routes and interfaces configured for VNI 2000. Any traffic destined to 20.0.0.3 will be dropped in the kernel since there are no routes or interfaces configured for VxLAN 2000. 
+
+Moreover, if the CPU port is set to Egress mode in the NPU, the packets sent from the CPU are directed to the egress pipeline. Hence, the kernel has to form the packets with appropriate VxLAN headers before placing the packet in the egress pipeline. To encap the packets with VxLAN headers, the kernel should have the VxLAN interface and routes configured.
```

Comment:
please remove this section about CPU port in 'egress mode'. This can be confusing and deviating the original intent. The line above is sufficient